ext/standard/string.c

```diff
@@ -2133,26 +2133,12 @@ static zend_string *php_chunk_split(const char *src, size_t srclen, const char *
 	const char *p;
 	size_t chunks; /* complete chunks! */
 	size_t restlen;
-	size_t out_len;
 	zend_string *dest;
 
 	chunks = srclen / chunklen;
 	restlen = srclen - chunks * chunklen; /* srclen % chunklen */
 
-	if (chunks > INT_MAX - 1) {
-		return NULL;
-	}
-	out_len = chunks + 1;
-	if (endlen !=0 && out_len > INT_MAX/endlen) {
-		return NULL;
-	}
-	out_len *= endlen;
-	if (out_len > INT_MAX - srclen - 1) {
-		return NULL;
-	}
-	out_len += srclen + 1;
-
-	dest = zend_string_alloc(out_len * sizeof(char), 0);
+	dest = zend_string_safe_alloc(chunks+1, endlen, srclen+1, 0);
```

Comment:
This looks correct-ish, but for somewhat subtle reasons, so it would be good to add comments:

a) chunks + 1 can't overflow, because chunks is limited by ZEND_LONG_MAX, which is smaller than SIZE_MAX.
b) srclen + 1 can't overflow, because the maximum string length is smaller than SIZE_MAX due to the string header.

One caveat is that a) only holds if sizeof(zend_long) == sizeof(size_t), which is not true under the x32 ABI, where ZEND_LONG_MAX is larger than SIZE_MAX. However, we already have many other edge-case issues under x32 (e.g. the fact that we're truncating during the size_t cast here), so I'm okay with ignoring that.

Reply:
Should I explicitly mention the caveat for x32 in the comment?

Reply:
Can you please rebase over https://github.com/php/php-src/commit/d955ee9f6b7e60895cac412969f9ddec91fbd2bb? With that change, this should be just `zend_string_safe_alloc(chunks, endlen, srclen, 0)` and we don't need the comment anymore :)

Reply:
Done :D